### DISCUSSION TITLE
Publication with different profiles?

### DISCUSSION THREAD

**Discussion by gotson** [opds-community/drafts#70] [Q&A] [answered]
I am wondering if it's possible to represent a publication that confirms to multiple profiles.

For example a pdf can conform to the pdf profile, but also to the divina profile, where each page is converted to an image.

Can this be represented in opds 2?

Another case would be a comic book in epub format, that would confirm to both epub and divina profiles.

**Comment by mickael-menu** [comment]
I don't think this is possible, because a profile requires specific media types declared in the reading order.

What you could have, is a link with an `alternate` rel in `metadata` pointing to an alternate rendition/manifest.

Or declare a main profile (say, EPUB) and add in the reading order links an `alternate` bitmap resource for each XHTML one. But you need support from the reading system for that. This wouldn't fork for a PDF, because you have a single PDF resource but `n` bitmaps.

What's your use case exactly (user flow)? Maybe there's a better way to handle that.

**Reply by gotson** [reply]
Well it's more about client support. Komga currently handles everything as images, so a pdf would be served page by page converted in jpeg.

If a client cannot handle pdf, it's quite probably it can display images. Having alternate image resources (as a divina profile) would let the client display the pdf page by page, without the need to handle pdf format.

The same goes for epub that are image only, even though this use case is much more limited in practice. A client which cannot handle epub could still display images from a divina profile.

**Reply by mickael-menu** [reply]
This is usually handled as several acquisition links (with different media types) in the publication entry of the OPDS feed. The client app can choose which type of format it supports for the publication and download/stream just that.

**Reply by gotson** [reply]
Aren't acquisition links for download only?

The publication in the publications feed already contain the full publication detail, including the profile and resources.

**Reply by mickael-menu** [reply, marked as answer]
Not necessarily, if it points to a manifest it can be used for streaming. In Aldiko, we apply this approach to audiobooks. If a RWPM acquisition link is provided, we stream it. If a packaged audiobook acquisition link is available, we download it. In the user interface, we provide the option to switch between the manifest and the packaged version directly from the bookshelf. This is appreciated for offline reading.

Theoretically you could have a full publication (with `readingOrder`) in an OPDS 2 feed, but I don't know any implementation supporting this. It brings the issues we talked about with multiple profiles and the JSON payload of the feed might get quite large.

**Reply by gotson** [reply]
> Theoretically you could have a full publication (with `readingOrder`) in an OPDS 2 feed, but I don't know any implementation supporting this. It brings the issues we talked about with multiple profiles and the JSON payload of the feed might get quite large.

Thanks Mickael ! You managed to shed some light on a point that is very much _not clear at all_ in the spec, and that will really help me !
